src/test/compile-fail/feature-gate-nested_impl_trait.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+#![feature(conservative_impl_trait, universal_impl_trait)]
+
+use std::fmt::Debug;
+
+fn fine(x: impl Into<u32>) -> impl Into<u32> { x }
```

Comment:
can we add a test for `fn bad(x: impl Into<impl Debug>)` -- it occurs to me that nested syntax in this position is also "not great" for the same reason. Though perhaps a *mite* less bad in that it's more clear how we would allow those types to be specified in the future.

Reply:
Oh, I purposefully made that work, so that test would fail. I will make the change.

Reply:
Added.